src/shipit_uplift/shipit_uplift/coverage_by_changeset_impl.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import requests
+import whatthepatch
+from shipit_uplift.coverage import coverage_service
+
+
+def get_coverage_builds(changeset):
+    '''
+    This function returns the last successful build before a given
+    changeset and the first successful coverage build after the same
+    changeset.
+    '''
+    r = requests.get('https://hg.mozilla.org/mozilla-central/json-rev/%s' % changeset)
+    push_id = int(r.json()['pushid'])
+
+    # In a span of 15 pushes, we hope we will find successful coverage builds.
+    r = requests.get('https://hg.mozilla.org/mozilla-central/json-pushes?startID=%s&endID=%s' % (push_id - 8, push_id + 7))
```

Comment:
We could perhaps teach the Coverage classes to keep track which push IDs have coverage.

15 pushes should be pretty good but if we get a regression introduced on mozilla-central it might take several days before we land a fix.
This is not a stopper. If you have any thoughts on what "ideal" would look like we can file it and deal it with it later (assuming it becomes something that happens often enough).

Reply:
I will file an issue to track this.

Reply:
https://github.com/mozilla-releng/services/issues/605